server/src/test/java/org/elasticsearch/ingest/IngestServiceTests.java

```diff
@@ -1877,15 +1908,13 @@ public void testResolveRequiredOrDefaultPipelineDefaultPipeline() {
 
         // index name matches with IDM:
         IndexRequest indexRequest = new IndexRequest("idx");
-        boolean result = IngestService.resolvePipelines(indexRequest, indexRequest, metadata);
-        assertThat(result, is(true));
```

Comment:
It seems me like you're testing less than we used to -- all these lost `assertThat(result, is(true));` and `assertThat(result, is(false));` should instead be testing `indexRequestHasPipeline` now, right?

Reply:
That test is in line 1887 actually. but yeah, I'd add expand these tests with calls to that method